backend/src/main/java/com/example/backend/service/ChatService.java

```diff
@@ -90,4 +88,11 @@ public void afterConnectionClosed(WebSocketSession session, CloseStatus status)
         this.session.remove(session);
     }
 
+    public List<ChatMessage> getPreviousMessages(String senderUsername, String receiverUsername){
+        List<ChatMessage> messages = chatRepo.findAllBySenderUsernameAndReceiverUsername(senderUsername,receiverUsername);
+        messages.addAll(chatRepo.findAllBySenderUsernameAndReceiverUsername(receiverUsername,senderUsername));
```

Comment:
Not quite sure if there is a better solution for this to get the chat history, but that's exactly how I did it.